Simplenote/NoticePresenter.swift

```diff
@@ -0,0 +1,170 @@
+import UIKit
+
+class NoticePresenter: KeyboardObservable {
+
+    // MARK: Properties
+    //
+    private var containerView = PassthruView(frame: .zero)
+    var noticeView: NoticeView?
+    private var noticeVariableConstraint: NSLayoutConstraint?
+
+    var isPresenting: Bool {
+        noticeView != nil
+    }
+    private var keyboardHeight: CGFloat = .zero
+    private var keyboardFloats: Bool = false
+    private var keyboardNotificationTokens: [Any]?
+    var timer: Timer?
+
+
+    // MARK: Lifecycle
+    //
+    deinit {
+        stopListeningToKeyboardNotifications()
+    }
+
+    // MARK: Presenting/Dismissing Methods
+    //
+    func presentNoticeView(_ noticeView: NoticeView, completion: @escaping () -> Void) {
+        guard let keyWindow = getKeyWindow() else {
+            return
+        }
+        self.noticeView = noticeView
+
+        containerView.addSubview(noticeView)
+        keyWindow.addFillingSubview(containerView)
+
+        noticeVariableConstraint = noticeView.topAnchor.constraint(equalTo: containerView.bottomAnchor)
+        noticeVariableConstraint?.isActive = true
+        noticeView.centerXAnchor.constraint(equalTo: containerView.centerXAnchor).isActive = true
+        containerView.layoutIfNeeded()
+
+        displayNotificationView() {
```

Comment:
`noticeVariableConstraint` is set in L38 and immediately after, it's replaced in L71.

You can probably drop L37 + L38 + L40

Reply:
Lines `L37 + L38 + L40` are here to set the initial offscreen position with top side of notice attached to bottom of container.

Reply:
Discussed over Slack with Evgeny, and indeed, the solution I was proposing to animate things relies on the `intrinsicContentSize` calculation, which is not 100% bullet proof.

Please disregard my comment Charlie!! (and thank you @eshurakov !!)